diff --git a/getopt.c b/getopt.c
index 9d29de7..c71f955 100644
--- a/getopt.c
+++ b/getopt.c
@@ -135,7 +135,7 @@ static struct _getopt_data getopt_data;
    whose names are inconsistent.  */
 
 #ifndef getenv
-extern char *getenv ();
+extern char *getenv (const char *);
 #endif
 
 #endif /* not __GNU_LIBRARY__ */
diff --git a/getopt.h b/getopt.h
index 7684fc4..aa53837 100644
--- a/getopt.h
+++ b/getopt.h
@@ -166,7 +166,7 @@ extern int __posix_getopt (int ___argc, char *const *___argv,
 #  endif
 # endif
 #else /* not __GNU_LIBRARY__ */
-extern int getopt ();
+extern int getopt (int ___argc, char *const *___argv, const char *__shortopts);
 #endif /* __GNU_LIBRARY__ */
 
 #ifndef __need_getopt
